File: velocity/src/main/java/rocks/milspecsg/msessentials/commands/UnBanCommand.java
package rocks.milspecsg.msessentials.commands;

import com.google.inject.Inject;
import com.velocitypowered.api.command.Command;
import com.velocitypowered.api.command.CommandSource;
import net.kyori.text.TextComponent;
import org.checkerframework.checker.nullness.qual.NonNull;
import rocks.milspecsg.msessentials.api.member.MemberManager;
import rocks.milspecsg.msessentials.modules.messages.CommandUsageMessages;
import rocks.milspecsg.msessentials.modules.messages.PluginMessages;
import rocks.milspecsg.msessentials.modules.utils.PluginPermissions;

public class UnBanCommand implements Command {

    @Inject
    private PluginMessages pluginMessages;

    @Inject
    private MemberManager<TextComponent> memberManager;

    @Inject
    private CommandUsageMessages commandUsage;

    @Override
    public void execute(CommandSource source, @NonNull String[] args) {
        if (!source.hasPermission(PluginPermissions.BAN)) {
            source.sendMessage(pluginMessages.noPermission);
            return;
        }
        if (!(args.length > 0)) {
            source.sendMessage(pluginMessages.notEnoughArgs);
            source.sendMessage(commandUsage.unbanCommandUsage);
            return;
        }
        memberManager.unBan(args[0]).thenAcceptAsync(source::sendMessage);
    }
}

File: common/src/main/kotlin/org/anvilpowered/catalyst/common/command/UnBanCommand.kt
/*
 *     Copyright (C) 2020 STG_Allen
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package org.anvilpowered.catalyst.common.command

import com.google.inject.Inject
import com.mojang.brigadier.context.CommandContext
import org.anvilpowered.anvil.api.registry.Registry
import org.anvilpowered.anvil.api.util.PermissionService
import org.anvilpowered.anvil.api.util.TextService
import org.anvilpowered.catalyst.api.member.MemberManager
import org.anvilpowered.catalyst.api.plugin.PluginMessages
import org.anvilpowered.catalyst.api.registry.CatalystKeys

class UnBanCommand<TString, TCommandSource> @Inject constructor(
  private val permissionService: PermissionService,
  private val textService: TextService<TString, TCommandSource>,
  private val pluginMessages: PluginMessages<TString>,
  private val memberManager: MemberManager<TString>, 
  private val registry: Registry,
){
  fun execute(source: TCommandSource, args: Array<String>) {
    if (!permissionService.hasPermission(source, registry.getOrDefault(CatalystKeys.BAN_PERMISSION))) {
      textService.send(pluginMessages.noPermission, source)
      return
    }
    if (args.isEmpty()) {
      textService.send(pluginMessages.notEnoughArgs, source)
      textService.send(pluginMessages.unbanCommandUsage(), source)
      return
    }
    memberManager.unBan(args[0]).thenAcceptAsync { textService.send(it, source) }
  }

  fun unban(context: CommandContext<TCommandSource>): Int {
    execute(context.source, arrayOf(context.getArgument("target")))
    return 1
  }
}
